server.js

```diff
@@ -42,9 +42,56 @@ server.use('/dev', proxy(DEVSERVER))
 server.listen(PORT, () => console.log(`listening on port ${PORT}`))
 
 
+const read_active_filter = ({on_filter}) => {
+    // Step 1: See if there is an active context
+    const tw_context = readline.createInterface({
+        input: spawn_tw(['_get', 'rc.context']).stdout
+    })
+    let context = ''
+    tw_context.on('line', (line) => {
+        if (!context && line) context = line
+    })
+    tw_context.on('close', (line) => {
+        // Step 2: If there is an active context, get its filter.
+        //         If not, get the filter for the "next" report.
+        let cmd = ['_get']
+        if (context)
+            cmd.push('rc.context.' + context)
+        else
+            cmd.push('rc.report.next.filter')
+
+        const tw_filter = readline.createInterface({
+            input: spawn_tw(cmd).stdout
+        })
+
+        let filter = ''
+        tw_filter.on('line', (line) => {
+            if (!filter && line) filter = line
+        })
+        tw_filter.on('close', (line) => {
+            // Execute callback with obtained filter
+            on_filter('(' + filter + ')')
+        })
+    })
+}
+
+
+const export_filtered_tasks = (filter, callbacks) => {
+    // If a filter is already passed, don't bother checking taskwarrior's
+    if (filter)
+        return export_tasks(filter, callbacks)
+
+    read_active_filter({on_filter: (f) => export_tasks(f, callbacks)})
+}
+
+
 const export_tasks = (filter, {on_data, on_exit}) => {
-    const f = `${ALWAYS_FILTER} ${filter || ''}`
-    const tw = spawn(TASK, TASKOPTS.concat(['rc.json.array=on', f, 'export']))
+    let cmd = ['rc.json.array=on']
+    if (filter)
+        cmd.push(filter)
+    cmd.push(ALWAYS_FILTER)
+    cmd.push('export')
+    const tw = spawn_tw(cmd)
```

Comment:
I think my previous code was more readable than spreading this over multiple lines. Well, not really, but `let cmd = ['rc.json.array=on', ALWAYS_FILTER, filter || '', 'export']` would be :D But maybe that's just me, feel free to ignore.

Reply:
Remember I mentioned having an empty string as argument broke everything in a weird way :-p so that doesn't really work.

Reply:
oh :D okay, sorry :D